netx/net/sourceforge/jnlp/util/XDesktopEntry.java

```diff
@@ -477,13 +485,98 @@ private void cacheIcon() throws IOException, NonFileProtocolException {
             if (targetName.equals(FAVICON)) {
                 targetName = file.getNotNullProbalbeCodeBase().getHost() + ".ico";
             }
-            File target = new File(PathsAndFiles.ICONS_DIR.getFile(), targetName);
-            Files.copy(source.toPath(), target.toPath(), StandardCopyOption.REPLACE_EXISTING);
+            File target = null;
```

Comment:
bad pattern.

define target as `final` in first if and move code of 2nd if to first one.

Reply:
When I do that, the code on will not work on linux anymore.

Reply:
this is related to the redefinition of `target`. In that case you need to use 2 `final` variables.